src/shapes/text.class.js

```diff
@@ -844,13 +854,34 @@
      */
     _setGraphemeOnPath: function(positionInPath, graphemeInfo, startingPoint) {
       var centerPosition = positionInPath + graphemeInfo.kernedWidth / 2,
-          path = this.path;
+          path = this.path, offsetDist = 0;
 
       // we are at currentPositionOnPath. we want to know what point on the path is.
       var info = fabric.util.getPointOnPath(path.path, centerPosition, path.segmentsInfo);
       graphemeInfo.renderLeft = info.x - startingPoint.x;
       graphemeInfo.renderTop = info.y - startingPoint.y;
       graphemeInfo.angle = info.angle + (this.pathSide ===  'right' ? Math.PI : 0);
+
+      if (this.type === 'text' && this.path) {
+        switch (this.pathAlign) {
+          case 'center':
+            offsetDist = graphemeInfo.height / 4;
+            break;
+          case 'ascender':
+            offsetDist = graphemeInfo.height / 1.5;
+            break;
+          case 'descender':
+            offsetDist = graphemeInfo.height / -3.75;
+            break;
+        }
+      }
+
+      if (offsetDist) {
+        var angle = graphemeInfo.angle - Math.PI / 2;
```

Comment:
Subtracting math.pi/2 does not explain why we are doing that, and can be probably be done swapping sin/cos between them ( and maybe changing the cos sign too )

Reply:
you're right, that is much cleaner